File: BehaviorPattern/Observer/Golang/Concretes/Peashooter.go
package concretes

import (
	"v/Interfaces"
	"fmt"
)

type Peashooter struct {
	PlantAction string
}

// Constructor
func NewPeashooter() interfaces.Observer {
	return &Peashooter{}
}

// Overise update
func (p *Peashooter) Update(action string) {
	p.PlantAction = action
	fmt.Println("Peashooter: Zombie action detected: " + action)
}

File: BehaviorPattern/Observer/Java/Concretes/Peashooter.java
package Concretes;

import Interfaces.Observer;

public class Peashooter implements Observer {
    private String plantAction;

    @Override
    public void update(String action) {
        this.plantAction = action;
        System.out.println("Peashooter: Zombie action detected: " + action);
    }
}
